internal/service/resources/cmd/serve.go

```diff
@@ -1,25 +1,98 @@
 package cmd
 
 import (
+	"fmt"
 	"log/slog"
 	"os"
 
-	"github.com/openshift-kni/oran-o2ims/internal/service/resources"
 	"github.com/spf13/cobra"
+
+	"github.com/openshift-kni/oran-o2ims/internal/cmd/server"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	"github.com/openshift-kni/oran-o2ims/internal/service/resources"
+	"github.com/openshift-kni/oran-o2ims/internal/service/resources/api"
 )
 
+// config defines the configuration attributes for the resource server
+var config api.ResourceServerConfig
+
 // resourceServer represents start command for the resource server
 var resourceServer = &cobra.Command{
 	Use:   "serve",
 	Short: "Start resource server",
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := resources.Serve(); err != nil {
+		if err := resources.Serve(&config); err != nil {
 			slog.Error("failed to start resource server", "err", err)
 			os.Exit(1)
 		}
 	},
 }
 
+// setServerFlags creates the flag instances for the server
+func setServerFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.StringVar(
+		&config.Listener.Address,
+		server.APIListenerAddressFlagName,
+		"127.0.0.1",
+		"API listener address.",
+	)
+	flags.StringVar(
```

Comment:
Correct, it is currently not needed because we terminate our TLS at the RBAC proxy, but given the recent announcement about the proxy and kubebuilder is quite likely that we'll have to move that back to the server directly.  Still a bit TBD.

Reply:
removed